Sprint-2/two-querystring.js

```diff
@@ -0,0 +1,54 @@
+/*function parseQueryString(queryString) {
+  const queryParams = {};
+
+  const [key, value] = queryString.split("="); // key will hold 'fruit', value will hold 'banana'
+  queryParams[key] = value;
+
+  return queryParams;
+}*/
+
+/*function parseQueryString(queryString) {
+  const queryParams = {};
+
+  const keyValuePair = queryString.split("=");
+  const key = keyValuePair[0]; // key will hold 'fruit'
+  const value = keyValuePair[1]; // value will hold 'banana'
+  queryParams.key = value;
+
+  return queryParams;
+}*/
+
+/*function parseQueryString(queryString) {
+  const queryParams = {};
+  if (queryString.length === 0) {
+    return queryParams;
+  }
+
+  const [key, value] = queryString.split("="); // will hold ['fruit', 'banana']
+  queryParams[key] = value; // will set the property name with the value of the key variable
+
+  return queryParams;
+}*/
+
+function parseQueryString(queryString) {
+  const queryParams = {}; //This will store the key-value pairs from the query string.
+  if (queryString.length === 0) {
+    return queryParams; //If the input string is
+    //empty (""), we return the empty object right away because there are no query parameters to process.
+  }
+  const keyValuePairs = queryString.split("&"); //split("&") to break the query string
+  // into smaller parts, separating them at every &.
+
+  //The for loop processes each pair in the array, such as "name=John".
+   for (const pair of keyValuePairs) {
+     const [key, value] = pair.split("="); //Split Each Pair: pair.split("=")
+     //breaks the string into key and value
+     queryParams[key] = value; //queryParams[key] = value stores the key-value
+     //pair in the queryParams object.After the loop finishes, queryParams contains
+     // all the key-value pairs, and the function returns it.
+   }
+
+  return queryParams;
+}
```

Comment:
**Todo**
For each of the following function calls, does your function return the value you expected?
```
parseQueryString("a=b&=&c=d")
parseQueryString("a=")
parseQueryString("=b")
parseQueryString("a=b&&c=d")
parseQueryString("a%25b=c%26d`)
```
Note: the `%25` and `%26` are ***URL encoded*** or ***percent encoded*** characters.

Reply:
Hi Cyuan, 
I did make some changes but honestly, it is another level to me. I'm studying these days JS things I kow I left behind and theses tests are one of them.